c10/core/Scalar.h

```diff
@@ -158,14 +198,68 @@ class C10_API Scalar {
     }
   }
 
+  Scalar(Scalar&& rhs) noexcept : tag(rhs.tag) {
+    moveFrom(std::move(rhs));
+  }
+
+  Scalar(const Scalar& rhs) : Scalar(rhs.v, rhs.tag) {}
+
+  Scalar(c10::SymInt si) : tag(Tag::HAS_si) {
+    v.si = si;
+  }
+
+  C10_ALWAYS_INLINE void moveFrom(Scalar&& rhs) noexcept {
+    if (rhs.tag == Tag::HAS_si) {
+      new (&v.si) c10::SymInt(rhs.v.si);
+      rhs.v.si.release_();
+    } else {
+      v.u = rhs.v.u;
+    }
+    tag = rhs.tag;
+    rhs.clearToInt();
+  }
+
+  void clearToInt() noexcept {
+    v.u.i = 0;
+    tag = Tag::HAS_i;
+  }
+
+  // We can't set v in the initializer list using the
```

Comment:
a random comment, I think struct initialization only works if a struct doesn't have any c-tors :) the other compilers must be more lenient than msvc :)